sqlmesh/core/context.py

```diff
@@ -680,10 +680,13 @@ def load(self, update_schemas: bool = True) -> GenericContext[C]:
                 cache_dir=self.cache_dir,
             )
 
-            models = self.models.values()
-            for model in models:
-                # The model definition can be validated correctly only after the schema is set.
-                model.validate_definition()
+            # The model definition can be validated correctly only after the schema is set.
+            for model in self.models.values():
+                if isinstance(model, SqlModel):
+                    # Validates only the model metadata; SQL validations are handled by the linter
+                    super(SqlModel, model).validate_definition()  # type: ignore
```

Comment:
For whatever reason, mypy wasn't happy without this `# type: ignore`:

```
sqlmesh/core/context.py:687: error: "validate_definition" undefined in superclass  [misc]
Found 1 error in 1 file (checked 515 source files)
```

Seems like its own limitation though, since `_Model` clearly defines `validate_definition`.

Reply:
i think you should keep the old version, we still call validate_definition() for somethings, but just move some of the logic out

Reply:
validation_definition should be defined, some change of your must have messed with the types?

Reply:
> validation_definition should be defined, some change of your must have messed with the types?

This code path is definitely hit at runtime, so this should work. Don't see how my changes could affect types.

> I think you should keep the old version, we still call validate_definition() for somethings, but just move some of the logic out

Ah, so you prefer just picking out the duplicate projection / aliasing validations and moving them here?

Reply:
Ok based on https://github.com/TobikoData/sqlmesh/pull/5305#discussion_r2326063423 it seems like that's it. I'll give it a whirl.